states/statemgr/filesystem.go

```diff
@@ -100,6 +125,23 @@ func (s *Filesystem) WriteState(state *states.State) error {
 
 	defer s.mutex()()
```

Comment:
hehe, I didn't notice your little mutex wrapper before ;p I've flip-flopped in this `()()` pattern a couple times, but if we use it more often it will become more recognizable to other readers.

Reply:
I don't think this was my original thought here... not sure where it came from, but I'm pretty sure I just copied it from somewhere else when working on this!